examples/throughput/subscriber.cpp

```diff
@@ -0,0 +1,327 @@
+}
+
+template<typename T>
+bool wait_for_writer(dds::sub::DataReader<T> &reader)
+{
+  std::cout << "\n" << subprefix << "Waiting for a writer ...\n" << std::flush;
+
+  dds::core::cond::StatusCondition sc = dds::core::cond::StatusCondition(reader);
+  sc.enabled_statuses(dds::core::status::StatusMask::subscription_matched());
+
+  dds::core::cond::WaitSet waitset;
+  waitset.attach_condition(sc);
+
+  dds::core::cond::WaitSet::ConditionSeq conditions =
+    waitset.wait(dds::core::Duration::from_secs(30));
+
+  for (const auto & c:conditions)
+  {
+    if (c == sc)
+      return true;
+  }
+
+  std::cout << subprefix << "Did not discover a writer.\n" << std::flush;
+
+  return false;
+}
+
+template<typename T>
+class ThroughputListener: public dds::sub::DataReaderListener<T>
+{
+  public:
+  /*implementation of virtual functions*/
+
+  /*only on_data_available does anything*/
+  void on_data_available(dds::sub::DataReader<T>& rd) {
+    (void)do_take(rd);
+  }
+
+  /*all others are just dummies*/
+  void on_requested_deadline_missed(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::RequestedDeadlineMissedStatus&) { }
+
+  void on_requested_incompatible_qos(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::RequestedIncompatibleQosStatus&) { }
+
+  void on_sample_rejected(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::SampleRejectedStatus&) { }
+
+  void on_liveliness_changed(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::LivelinessChangedStatus&) { }
+
+  void on_subscription_matched(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::SubscriptionMatchedStatus&) { }
+
+  void on_sample_lost(
+      dds::sub::DataReader<T>&,
+      const dds::core::status::SampleLostStatus&) { }
+};
+
+int main (int argc, char **argv)
+{
+
+  if (parse_args(argc, argv) == EXIT_FAILURE)
+  {
+    return EXIT_FAILURE;
+  }
+
+  std::cout << subprefix << "Cycles: " << maxCycles << " | PollingDelay: " << pollingDelay.count() << " ms | Partition: " << partitionName << "\n"
+            << subprefix << "Using a " << (pollingDelay.count() > 0 ? "polling" : pollingDelay.count() < 0 ? "listener" : "waitset") << " approach.\n" << std::flush;
+
+  dds::domain::DomainParticipant participant(domain::default_id());
+
+  dds::topic::Topic<ThroughputModule::DataType> topic(participant, "Throughput");
+
+  dds::sub::qos::SubscriberQos sqos;
+  sqos.policy(dds::core::policy::Partition(partitionName));
+
+  dds::sub::Subscriber subscriber(participant, sqos);
+
+  dds::sub::qos::DataReaderQos rqos;
+  rqos.policy(
+    dds::core::policy::Reliability(
+      dds::core::policy::ReliabilityKind::Type::RELIABLE,
+      dds::core::Duration::from_secs(10)));
+  rqos.policy(
+    dds::core::policy::History(
+      dds::core::policy::HistoryKind::Type::KEEP_ALL,
+      0));
+  rqos.policy(
+    dds::core::policy::ResourceLimits(MAX_SAMPLES));
+
+  ThroughputListener<ThroughputModule::DataType> list;
```

Comment:
The name "list" for a listener might be a bit confusing. Ususally you mean something else when referring to a list.

Reply:
Maybe abbreviated the variable name a little too much here :)
I will look at all variable names and make them a little more explicit.